### PR TITLE
Nitryl Gloves reduce firefighter carry delay (tg#46735)

### PR DESCRIPTION
**Pull request by Cenrus**
Re-PR of #559, It didn't get merged because I had no idea what I was doing.

Makes fireman carrying more useful for paramedics because they have access to nitryl gloves.

:cl: carlarctg
add: Reduced firefight carry delay with latex or nitrile gloves
/:cl:

### CODE REVIEW

**Comment by ToasterBan** [comment]
as someone who likes to play doctor



PLEASE MERGE

File: code/modules/clothing/gloves/color.dm
Comment (park66665) [inline code/modules/clothing/gloves/color.dm:189]: Maybe do the same with obj_break?
```suggestion
/obj/item/clothing/gloves/color/latex/dropped(mob/user)
	..()
	REMOVE_TRAIT(user, carrytrait, CLOTHING_TRAIT)

/obj/item/clothing/gloves/color/latex/obj_break(mob/user)
	..()
	REMOVE_TRAIT(user, carrytrait, CLOTHING_TRAIT)
```

File: code/modules/mob/living/carbon/human/human.dm
Comment (park66665) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
	if(HAS_TRAIT(src, TRAIT_QUICKER_CARRY))
		carrydelay = 30
		skills_space = " expertly"
	else if(HAS_TRAIT(src, TRAIT_QUICK_CARRY))
		carrydelay = 40
		skills_space = " quickly"
	if(can_be_firemanned(target) && !incapacitated(FALSE, TRUE))
		visible_message("<span class='notice'>[src] starts[skills_space] lifting [target] onto their back..</span>",
		//Joe Medic starts quickly/expertly lifting Grey Tider onto their back..
		"<span class='notice'>[HAS_TRAIT(src, TRAIT_QUICKER_CARRY) ? "Using your gloves' nanochips, you" : "You"][skills_space] start to lift [target] onto your back[HAS_TRAIT(src, TRAIT_QUICK_CARRY) ? ", while assisted by the nanochips in your gloves..." : "..."]</span>")
```

(Spelling)

File: code/__DEFINES/traits.dm
Comment (park66665) [inline code/__DEFINES/traits.dm:158]: Don't forget to add those to code/_globalvars/traits.dm

File: code/modules/clothing/gloves/color.dm
Comment (park66665) [inline code/modules/clothing/gloves/color.dm]:
```suggestion
/obj/item/clothing/gloves/color/latex/obj_break()
	..()
	if(ishuman(loc))
		REMOVE_TRAIT(loc, carrytrait, CLOTHING_TRAIT)
```

Someone merge this if the author reads this and decides to commit it because I am going to sleep soon(tm)

**Comment by jupyterkat** [comment]
Wait for skillchips and add the trait to that instead

File: code/modules/mob/living/carbon/human/human.dm
Comment (utilbird) [inline code/modules/mob/living/carbon/human/human.dm:1074]: dupe check?

**Comment by Fronsis** [comment]
I've been begging for this feature to be merged for so many months

BLEASE MERGE

File: code/modules/clothing/gloves/color.dm
Comment (Archanial) [inline code/modules/clothing/gloves/color.dm:180]: Do you need var for this?

File: code/modules/mob/living/carbon/human/human.dm
Comment (Archanial) [reply code/modules/mob/living/carbon/human/human.dm:1074]: Those are two different trait, quick and quicker carry.

File: code/modules/clothing/gloves/color.dm
Comment (park66665) [reply code/modules/clothing/gloves/color.dm:180]: If you don't use a var, then there should be another set of overrides for `/obj/item/clothing/gloves/color/latex/nitrile` which actually will be 3 times more messier.

**Comment by park66665** [comment]
Me: Takes a final look
The CL: "Halves"
Me:
Stare

File: code/modules/clothing/gloves/color.dm
Comment (Archanial) [reply code/modules/clothing/gloves/color.dm:180]: right, but it should at least follow naming scheme and be called carry_trait

**Comment by DatBoiTim** [comment]
> Wait for skillchips and add the trait to that instead

Genius. - The Author of Skillchips

**Review by park66665 — APPROVED** [review]
If you want this skillchip'd, then argue for it successfully and then code it after #3242 actually gets merged and becomes stable and well-accepted. Or, of course, you can include the thing in #3242 to accelerate the process.

All clear. This PR is good to go.

**Comment by DatBoiTim** [comment]
> If you want this skillchip'd, then argue for it successfully and then code it after #3242 actually gets merged and becomes stable and well-accepted. Or, of course, you can include the thing in #3242 to accelerate the process.
> 
> All clear. This PR is good to go.

Hahaha @park66665 you assume I won't code this into a skillchip right now >:D

**Comment by jupyterkat** [comment]
Tg already made this into skillchips, add it in your skillchips and close this one.
tgstation/tgstation#56593

**Comment by DatBoiTim** [comment]
> Tg already made this into skillchips
> [tgstation/tgstation#56593](https://github.com/tgstation/tgstation/pull/56593)

I'm doing my own implementation cause I'm stubborn also because it was a bounty with a unique implementation roadmap

**Comment by jupyterkat** [comment]
Reinventing the wheel for the 50th time...

**Comment by DatBoiTim** [comment]
> Reinventing the wheel for the 50th time...

Yes

**Comment by park66665** [comment]
PRs are not merged until they are merged. You can't guarantee the skillchip PR's future enough to warrant the closure of this PR. This generally applies to any PR, really.

**Comment by DatBoiTim** [comment]
> PRs are not merged until they are merged. You can't guarantee the skillchip PR's future enough to warrant the closure of this PR. This generally applies to any PR, really.

Oh I am aware, however, I am adding this as we speak

**Comment by park66665** [comment]
It would be bold to assume whether you want this PR closed or not. So, let me make things clear. You are free to code skillchip version of this concept, and I am free to merge this. Moreover, I *am* going to merge this in order to avoid the worst outcome and reduce the future decision fatigue and problems. You may blame me for not making the best decision, but "the best decision" became impossible the moment you decided to become stubborn.

You are expected to utilize the traits and the code changes from this PR if you want skillchips to shorten the fireman carry delay.

**Comment by DatBoiTim** [comment]
> It would be bold to assume whether you want this PR closed or not. So, let me make things clear. You are free to code skillchip version of this concept, and I am free to merge this. Moreover, I _am_ going to merge this in order to avoid the worst outcome and reduce the future decision fatigue and problems. You may blame me for not making the best decision, but "the best decision" became impossible the moment you decided to become stubborn.
> 
> You are expected to utilize the traits and the code changes from this PR if you want skillchips to shorten the fireman carry delay.

I can work with that. The stubbornness was more to the port of TGChips.

**Pull request merged**
